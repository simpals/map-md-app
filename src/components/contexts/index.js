import { createContext } from 'react';

export const MapCtx = createContext({
  MapGL: null,
  userLocation: null,
  authorization: null,
  style: null,
  changeStyle: () => {}
});
