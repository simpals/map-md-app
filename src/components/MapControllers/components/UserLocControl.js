import React, { useContext } from 'react';
import styled from 'styled-components';
import { MapCtx } from '../../contexts';

const UserLocationWrapper = styled.div`
  position: absolute;
  background-color: #ffffff;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.1), 0 3px 10px 0 rgba(0, 0, 0, 0.07);
  border-radius: 100px;
  width: 36px;
  height: 36px;
  top: 16px;
  right: 14px;
  & > * {
    display: inline-block;
    vertical-align: middle;
  }
  ${({ theme: { userLocation } }) => userLocation && userLocation}
`;

const MylocBtn = styled.button`
  padding: 9px 5px;
  display: inline-block;
  margin: 0;
`;

const ToolIcon = styled.span`
  font-size: 18px;
  line-height: 0;
  color: rgba(0, 0, 0, 0.54);
  &:hover,
  &.active {
    color: #ff7201;
  }
`;

const UserLocControl = () => {
  const { userLocation } = useContext(MapCtx);

  const getUserLocation = () => {
    userLocation.once('error', (e) => {
      window.alert('Position could not be determined.');
    });
    if (userLocation) {
      userLocation.trigger();
    }
  };

  return (
    <UserLocationWrapper className='UserLocControl' >
      <MylocBtn type='button' onClick={() => getUserLocation()}>
        <ToolIcon className='icon-near_me' />
      </MylocBtn>
    </UserLocationWrapper>
  );
};

export default UserLocControl;
