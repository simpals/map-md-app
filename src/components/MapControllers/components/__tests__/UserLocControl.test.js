import 'jsdom-global/register';
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserLocControl from '../UserLocControl';
import { MapCtx } from '../../../contexts';

const context = {
  userLocation: {
    once: jest.fn(),
    trigger: jest.fn()
  }
};

describe('test UserLocControl component', () => {
  it('renders without crashing', () => {
    const component = shallow(<UserLocControl />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('test UserLocation click', () => {
    const component = mount(
      <MapCtx.Provider value={context}>
        <UserLocControl />
      </MapCtx.Provider>
    );
    component.find('button').simulate('click');
  });
});
