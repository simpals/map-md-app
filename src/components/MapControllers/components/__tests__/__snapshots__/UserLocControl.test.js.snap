// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test UserLocControl component renders without crashing 1`] = `
<styled.div
  className="UserLocControl"
>
  <styled.button
    onClick={[Function]}
    type="button"
  >
    <styled.span
      className="icon-near_me"
    />
  </styled.button>
</styled.div>
`;
